src/vs/base/browser/ui/dialog/dialog.css

```diff
@@ -60,6 +60,7 @@
 /** Dialog: Message Container */
 .monaco-dialog-box .dialog-message-row .dialog-message-container {
 	display: flex;
+	flex: 1; /* let the dialog message contents grow */
```

Comment:
This lets the message area grow with the box, but the way it works right now is to grow with the conent. I'm not sure this change is necessary as I would expect the input to define that. E.g. all of the current text divs grow with long text until they cannot and then they wrap. Shouldn't this just be a a div with text stylized like an input? Side note, I noticed I can edit the text in the input when testing. Is that expected?

Reply:
> Shouldn't this just be a a div with text stylized like an input

The `input` api is designed for requesting passwords and other kinds of text input, so generally it needs to be editable. For my scenario I don't _need_ it to be editable, but I do want that the user can keyboard-nav through the text to see its contents in full, and I get that for free by using the `input` dialog. I don't think there is a way for the `input` element to specify that it should grow with its contents using CSS attributes alone though.